agent/util-scripts/pbench-tool-meister-start

```diff
@@ -451,6 +456,9 @@ class ToolDataSink(BaseServer):
                 kind = data["kind"]
                 action = data["action"]
                 status = data["status"]
+                if status != "success":
+                    tms = json.loads(redis_client.get(tms_tool_info))
+                    log_failed_tool_init(logger, tms)
```

Comment:
Given that we're succeeding or failing based on `status == "success"` at the end of the loop, and that we went to the trouble of declaring `status = ""` before the loop, I think we shouldn't be overwriting `status` from each iteration. Instead, maybe just start with `all_ok = True` and on each iteration, `if status != "success` set `all_ok` to False. Then at the end return 0 if `all_ok` else 1. (Or even `final_status = 0` and set it to 1 if any TM reported failure, in which case you can just return `final_status` at the end.)

Right now, `wait` (and the startup orchestration) is failing only if the *last* remote reported failure.

(This isn't new with your change, but this is probably a good time to fix it since you're extending the definition of "failed".)

Reply:
Sure :)